xmtp_mls/src/lib.rs

```diff
@@ -47,4 +47,54 @@ mod tests {
             tracing_subscriber::fmt::init();
         })
     }
+
+    /// wrapper over assert!(matches!()) for Errors
+    /// assert_err!(fun(), StorageError::Explosion)
+    ///
+    /// or the message variant,
+    /// assert_err!(fun(), StorageError::Explosion, "the storage did not explode");
+    #[macro_export]
+    macro_rules! assert_err {
+        ( $x:expr , $y:pat $(,)? ) => {
+            assert!(matches!($x, Err($y)));
+        };
+
+        ( $x:expr, $y:pat $(,)?, $($msg:tt)+) => {{
+            assert!(matches!($x, Err($y)), $($msg)+)
+        }}
+    }
+
+    /// wrapper over assert! macros for Ok's
+    ///
+    /// Make sure something is Ok(_) without caring about return value.
+    /// assert_ok!(fun());
+    ///
+    /// Against an expected value, e.g Ok(true)
+    /// assert_ok!(fun(), true);
+    ///
+    /// or the message variant,
+    /// assert_ok!(fun(), Ok(_), "the storage is not ok");
```

Comment:
Neat

Reply:
my goal here (as well as with `with_store` to make writing tests as ergonomic as possible. I did take some inspo from `tokio-test` and `frame-support` macros of the same name, but imo the more readable and easier it is to write tests the better